app/src/main/kotlin/edu/stanford/bdh/engagehf/contact/data/ContactDocumentToContactMapper.kt

```diff
@@ -0,0 +1,51 @@
+package edu.stanford.bdh.engagehf.contact.data
+
+import com.google.firebase.firestore.DocumentSnapshot
+import edu.stanford.spezi.core.design.component.StringResource
+import edu.stanford.spezi.modules.contact.model.Contact
+import edu.stanford.spezi.modules.contact.model.ContactOption
+import edu.stanford.spezi.modules.contact.model.PersonNameComponents
+import edu.stanford.spezi.modules.contact.model.call
+import edu.stanford.spezi.modules.contact.model.email
+import javax.inject.Inject
+
+class ContactDocumentToContactMapper @Inject constructor() {
+
+    fun map(document: DocumentSnapshot): Result<Contact> = runCatching {
```

Comment:
@eldcn Sounds good, I can do that! 

@pauljohanneskraft  @eldcn would it make sense from your point of view regarding iOS and in general to provide the optional values with a default = null value in the Contact Constructor, so that you don't have to explicitly set  
`description = null` etc. explicitly?

Reply:
There is a lot more sophisticated logic in iOS about the name components creation, though I'm fine with creating this very simple version, since we never really make use of the complexity of PersonNameComponents anyways - for the future, we may want to include a constructor in PersonNameComponents directly though taking a single string.

Regarding the default-nil thing: https://github.com/StanfordSpezi/SpeziContact/blob/main/Sources/SpeziContact/Models/Contact.swift this is already the case on iOS - must have missed this, sorry 😅